entrypoint.sh

```diff
@@ -100,37 +99,126 @@ function super_lint() {
   /action/lib/linter.sh
 }
 
+function get_files_matching_filetype() {
+  local matching_files=()
+  local filenames=("${@:2}")
+  local filetype=$1
+
+  for file in "${filenames[@]}"; do
+    filename=$(basename "$file")
+
+    if [[ "$filename" == *"$filetype" ]]; then
```

Comment:
nit: This reason that this works is because the `*` is a glob pattern, but using `=~` is better because it uses regex and we can also be explicit about the end of file with $, although there is no functional issue with what's currently there.

```suggestion
    if [[ "$filename" =~ .*${filetype}$ ]]; then
```

Reply:
Actually, let's stick with the globbing approach otherwise we'll need to escape all of the `.`s that you have in `filetype`